File: src/main/java/com/projectx/university/repository/StudentRepository.java
package com.projectx.university.repository;

import com.projectx.university.dto.Student;
import lombok.RequiredArgsConstructor;
import org.springframework.data.redis.core.ReactiveRedisOperations;
import org.springframework.stereotype.Repository;
import reactor.core.publisher.Mono;

@Repository
@RequiredArgsConstructor
public class StudentRepository {
    private final ReactiveRedisOperations<String, Student> redisTemplate;

    public Mono<Boolean> addStudent(Student student) {
        return redisTemplate.opsForValue().set(student.getId(), student);
    }

    public Mono<Student> getStudent(String studentId) {
        return redisTemplate.opsForValue().get(studentId);
    }
}

File: src/main/kotlin/com/projectx/university/repository/StudentRepository.kt
package com.projectx.university.repository

import com.projectx.university.dto.Student
import org.springframework.data.redis.core.ReactiveRedisOperations
import org.springframework.stereotype.Repository
import reactor.core.publisher.Mono

@Repository
class StudentRepository(private val redisTemplate: ReactiveRedisOperations<String, Student>) {
    fun addStudent(student: Student): Mono<Boolean> =
            redisTemplate.opsForValue().set(student.id, student)

    fun getStudent(studentId: String): Mono<Student> =
            redisTemplate.opsForValue().get(studentId)
}